h/services/group.py

```diff
@@ -21,11 +22,42 @@ def __init__(self, session, user_fetcher):
         self.session = session
         self.user_fetcher = user_fetcher
 
-    def fetch(self, pubid):
-        """Fetch a group by ``pubid``"""
+    def fetch(self, pubid_or_groupid):
+        """
+        Fetch a group, using either a groupid or a pubid
+
+        :param pubid_or_groupid: a string in either :mod:`~h.pubid` format
+                                 or as :attribute:`~h.models.group.Group.groupid`
+        :rtype:`~h.models.group.Group` or None
```

Comment:
This doesn't parse either, should be:

```
:rtype: :class:`~h.models.Group` or ``None``
```

A note about package structure btw: it's `h.models.Group` not `h.models.group.Group`. This is because `h/models/__init__.py` [exports all of its public interface as top-level package names](https://github.com/hypothesis/h/blob/7ea2f765f6b43a9fe14fbd5feb4093c7413683a2/h/models/__init__.py#L33).

Generally speaking code should be using `h.SOME_PACKAGE.*` rather than reaching in and using `h.SOME_PACKAGE.SOME_MODULE.*` because the internal structure of `SOME_PACKAGE` should be encapsulated within that package, should not be part of the package's public interface.

Packages need to use their `__init__.py` to define a public interface in this way and not all package's in h do so yet (for example `h.util` doesn't) but they should.

You could argue for an exception when a package is so big that exporting every public names as top-level `h.SOME_PACKAGE.*` names would be too long of a flat list of names or would have collisions, then it makes sense to structure the public interface of the package further. But in cases like that I'd first question whether the package is too big.